Cython/Compiler/ParseTreeTransforms.py

```diff
@@ -2915,6 +2915,11 @@ def visit_FuncDefNode(self, node):
         return node
 
     def visit_GILStatNode(self, node):
+        if node.condition is not None:
+            error(node.pos, "Non-constant condition in a "
+                            "`with %s(<condition>)` statement" % node.state)
```

Comment:
Nice. You could also use `node.condition.pos` instead of `node.pos` here, which then points directly to the non-constant condition.

Reply:
Changed to `node.condtion.pos`.